source/components/temp_doeclim_component.cpp

```diff
@@ -0,0 +1,528 @@
+/* Hector -- A Simple Climate Model
+   Copyright (C) 2014-2015  Battelle Memorial Institute
+
+   Please see the accompanying file LICENSE.md for additional licensing
+   information.
+*/
+/*
+ *  temp_doeclim_component.cpp
+ *  hector
+ *
+ *  Created by Ben Vega-Westhoff on 11/1/16.
+ *
+ */
+
+#include <boost/lexical_cast.hpp>
+#include <cmath>
+#include <limits>
+#define _USE_MATH_DEFINES
+
+#include "components/temp_doeclim_component.hpp"
+#include "core/core.hpp"
+#include "h_util.hpp"
+#include "models/simpleNbox.hpp"
+#include "visitors/avisitor.hpp"
+
+namespace Hector {
+  
+using namespace std;
+
+
+//------------------------------------------------------------------------------
+/*! \brief Constructor
+ */
+TempDOECLIMComponent::TempDOECLIMComponent() : internal_Ftot(0.0), last_Ftot(0.0) {
+}
+
+//------------------------------------------------------------------------------
+/*! \brief Destructor
+ */
+TempDOECLIMComponent::~TempDOECLIMComponent(){
+}
+    
+//------------------------------------------------------------------------------
+// documentation is inherited
+string TempDOECLIMComponent::getComponentName() const {
+    const string name = TEMP_DOECLIM_COMPONENT_NAME;
+    
+    return name;
+}
+
+
+//------------------------------------------------------------------------------
+/*! \brief              Calculates inverse of x and stores in y
+ *  \param[in] x        Assume x is setup like x = [a,b,c,d] -> x = |a, b|
+ *                                                                  |c, d|
+ *  \param[in] y        Inverted 1-d matrix
```

Comment:
This should be documented as `param[out]` or `param[inout]`, depending on whether the function expects `y` to be initialized.